### DISCUSSION TITLE
Database error when using SignUp and a trigger function

### DISCUSSION THREAD

**Discussion by jessie-mills-dev** [supabase/supabase#20041] [Questions] [answered]
I've got a component where I'm using the auth.signUp method to create a new user for my app. I'm using multi-tenancy and have instance_id tied to each table.

I'm sending through this data in my submit function:
```
const { error } = await supabaseAdmin.auth.signUp({
        email: newEmail,
        password: newPassword,
        options: {
          data: {
            first_name: newFirstName,
            last_name: newLastName,
            role: 'admin',
            email: newEmail,
            company_name: newInstanceName,
            instance_id: newInstanceId,
            plan_name: planName,
            seat_count: 1,
            free_trial_started: formattedToday,
            free_trial_last_day: formattedTrialLastDay,
            plan_status: 'trial',
          },
        },
      })
```
And I have a trigger function that adds the data sent through into various tables once it's been added to the auth.users one:

```
declare
  random_number text;
begin
  random_number := lpad(floor(random() * 999999)::text, 6, '0');
  
  insert into public.all_instances (company_name, instance_id)
  values (new.raw_user_meta_data->>'company_name', new.raw_user_meta_data->>'instance_id');
  insert into public.settings (id, instance_id, plan_name, plan_status, free_trial_started, free_trial_last_day, seat_count)
  values (new.id, new.raw_user_meta_data->>'instance_id', new.raw_user_meta_data->>'plan_name', new.raw_user_meta_data->>'plan_status', new.raw_user_meta_data->>'free_trial_started', new.raw_user_meta_data->>'free_trial_last_day', new.raw_user_meta_data->>'seat_count');
  insert into public.profiles (id, instance_id, role, email, first_name, last_name, url)
  values (new.id, new.raw_user_meta_data->>'instance_id', new.raw_user_meta_data->>'role', new.raw_user_meta_data->>'email', new.raw_user_meta_data->>'first_name', new.raw_user_meta_data->>'last_name', concat(new.raw_user_meta_data->>'first_name', '-', new.raw_user_meta_data->>'last_name', '-', random_number));
  return new;
end;

```
However I'm getting a database error - 500. I've been looking at this thread: https://github.com/supabase/supabase/issues/563 and I'm wondering if it's really just because I created the table via the dashboard vs SQL? I'm using that table in my app already and have a bunch of things relying on it and really don't want to recreate it if possible. 

Does anyone know what's going on here?

**Comment by GaryAustin1** [comment, marked as answer]
Look at the Postgres logs in the dashboard. It should show the exact error. 
Usually this is the function not being security definer type or when doing lots of fields a name or type mismatch.  You have to type text json to uuid or ints in most cases.

**Reply by jessie-mills-dev** [reply]
Thank you!! Looking at my logs properly helped - I found that it was a mismatch situation where the DB was expecting eg bigInt but I was submitting text. Followed the error trail and it's working.
